### PR TITLE
fix: soft delete

### PR DESCRIPTION
**Pull request by JimmyDaddy**
rt

### CODE REVIEW

File: lib/adapters/sequelize.js [hunk 1]
```diff
@@ -76,10 +76,11 @@ module.exports = Bone => {
     // EXISTS
     // static bulkCreate() {}
 
-    static async count(options = {}) {
+    static async count(options = {}, queryOptions = {}) {
       const { where, col, group } = options;
-      const spell = super.find(where);
+      let spell = super.find(where);
       if (Array.isArray(group)) spell.$group(...group);
+      if (queryOptions.paranoid === false || options.paranoid === false) spell = spell.unscoped;
```
Comment (cyjake) [inline lib/adapters/sequelize.js:83]: It seems there's no need to add an extra queryOptions. Sequelize treats options as the all in one options object, which is mostly handled by `translateOptions()` in our adapter.

File: lib/adapters/sequelize.js [hunk 2]
```diff
@@ -108,28 +109,32 @@ module.exports = Bone => {
...
       translateOptions(spell, options);
+      if (options.paranoid === false) spell = spell.unscoped;
       const [ rows, count ] = await Promise.all([ spell, spell.count() ]);
       return { rows, count };
     }
 
     static async findByPk(value) {
-      return await super.findOne({ [this.primaryKey]: value });
+      let spell = super.findOne({ [this.primaryKey]: value });
+      if (queryOptions.paranoid === false) spell = spell.unscoped;
```
Comment (cyjake) [inline lib/adapters/sequelize.js:136]: queryOptions is undefined here. `findByPk(value, options = {})` perhaps?

File: lib/adapters/sequelize.js [hunk 3]
```diff
@@ -108,28 +109,32 @@ module.exports = Bone => {
...
 
     static findAll(options = {}) {
-      const spell = this.find();
+      let spell = this.find();
       translateOptions(spell, options);
+      if (options.paranoid === false) spell = spell.unscoped;
       return spell;
     }
 
     static async findAndCountAll(options) {
```
Comment (cyjake) [inline lib/adapters/sequelize.js:126]: `options = {}`, in case `findAndCountAll()`

File: lib/adapters/sequelize.js [hunk 4]
```diff
@@ -151,12 +156,17 @@ module.exports = Bone => {
...
-      // findOne()
-      return this.findAll({ ...options, limit: 1 }).later(result => result[0]);
+      if (typeof options !== 'object') {
+        spell = super.findOne(options);
+      } else {
+        // findOne({ where })
+        // findOne()
+        spell = this.findAll({ ...options, limit: 1 }).later(result => result[0]);
+      }
+      if ((options && options.paranoid === false)) spell = spell.unscoped;
```
Comment (cyjake) [inline lib/adapters/sequelize.js:168]: just `return spell.unscoped;`

File: test/unit/adapters/sequelize.test.js [hunk 5]
```diff
@@ -191,6 +202,55 @@ describe('=> Sequelize adapter', () => {
     assert.throws(() => posts[0].content);
   });
 
+  it('Model.findAll(opt, { paranoid: false })', async () => {
```
Comment (cyjake) [inline test/unit/adapters/sequelize.test.js:205]: `findAll({ paranoid: false })` https://sequelize.org/master/class/lib/model.js~Model.html#static-method-findAll

File: test/unit/adapters/sequelize.test.js [hunk 6]
```diff
@@ -191,6 +202,55 @@ describe('=> Sequelize adapter', () => {
...
+      order: [[ 'createdAt', 'desc' ]],
+      limit: 1,
+    },  { paranoid: false });
+    assert.equal(posts.length, 1);
+    assert.equal(posts[0].title, 'Tyrael');
+
+    posts = await Post.findAll({
+      attributes: [ 'title' ],
+      where: { title: 'Leah' },
+    },  { paranoid: false });
```
Comment (cyjake) [inline test/unit/adapters/sequelize.test.js:248]:
```js
await Post.findAll({
  order: [[ 'createdAt', 'desc' ]],
  limit: 1,
  paranoid: false,
});
```

File: lib/bone.js [hunk 7]
```diff
@@ -345,9 +345,14 @@ class Bone {
     const where = { [primaryKey]: this[primaryKey] };
     if (shardingKey) where[shardingKey] = this[shardingKey];
 
-    return Bone.update.call(Model, where, changes).then(count => {
-      if (count === 1) this.syncRaw(changes);
-      return count;
+    const { updatedAt, deletedAt } = Model.timestamps;
```
Comment (JimmyDaddy) [inline lib/bone.js:348]: `instance.update`  should independent from `Class.update`, that will be more easy to implement hooks

**Pull request merged**